### PR TITLE
[Merged by Bors] - ci: Import `Archive` when checking YAML files

### PR DESCRIPTION
**Pull request by YaelDillies**
We sometimes want to refer to Archive declarations in the YAML files. This is currently impossible because the script in `checkYaml` only imports `Mathlib`. This PR changes it to also import `Archive`.

It also changes the hyperlinks to Archive to instead list declarations directly.

Also fix the fallout from #11530 that CI didn't catch that because we forgot to pipe the errors coming from `lake build Archive` (see #11612).

Co-authored-by: Enrico Borba <enricozb@gmail.com>

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
Can you prove this works by updating one of the existing `archive.yaml` entries to refer to the archive directly?

**Comment by eric-wieser** [comment]
(no need to do them all in this PR, though that would also be great!)

**Comment by eric-wieser** [comment]
bors d+

File: docs/100.yaml [hunk 1]
```diff
@@ -329,7 +329,7 @@
   author : Manuel Candales (archive), Michael Stoll (Mathlib)
 82:
   title  : Dissection of Cubes (J.E. Littlewood’s ‘elegant’ proof)
-  decl   : Theorems100.cannot_cube_a_cube
+  decl   : Theorems100.82.cannot_cube_a_cube
```
Comment (eric-wieser) [inline docs/100.yaml:332]: I think you need french quotes here.

Comment (YaelDillies) [reply docs/100.yaml:332]:
```suggestion
  decl   : Theorems100.«82».cannot_cube_a_cube
```

**Comment by YaelDillies** [comment]
bors merge

**Comment by YaelDillies** [comment]
bors -r

We seem to have hit a difference between CI for PRs and CI for master.

**Comment by YaelDillies** [comment]
bors merge

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**